.github/workflows/pre-commit.yml

```
@@ -4,10 +4,36 @@ on:
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"  # Set a consistent Python version
+
+      - name: Install dependencies
+        run: pip install --upgrade pip pre-commit  # Ensure latest pre-commit version
+
+      - name: Run pre-commit
+        run: pre-commit run --all-files --show-diff-on-failure || true  # Run all hooks without failing
+
+      - name: Check for changes and commit
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2006:style:1:7: Use $(...) notation instead of legacy backticks `...`

(shellcheck)